### PR TITLE
2.2 warn noproxy 1724338

### PR DESCRIPTION
**Pull request by jameinel**
## Description of change

LXD Provider should be warning on unknown config

LXD Provider was not calling ValidateUnknownAttrs because it wasn't
actually providing any new attributes. However, this meant that it
failed to catch any typos in the config values for core config.

This makes it so that "juju bootstrap lxd --config typo=foo" will warn
that 'typo' is not a known field. Also because it is very common to make
a mistake wrt '_' vs '-' I added a check to lookup the name by an
obvious alternative spelling, and give the hint to the user.

This also gets rid of all_go13.go because we have long since required go 1.3 for building Juju.

## QA steps

```
juju bootstrap lxd --model-default foo=bar --model-default http_proxy=123.123.123.123
WARNING unknown config field "http_proxy", did you mean "http-proxy"
WARNING unknown config field "foo"
```

This doesn't directly add tests, as it has to do with giving WARNING to the user.
We could potentially put this into CI tests.

## Documentation changes

None

## Bug reference

[lp:1724338](https://bugs.launchpad.net/juju/+bug/1724338)

### CODE REVIEW

File: environs/config/config.go [hunk 1]
```diff
@@ -1195,10 +1195,19 @@ func (cfg *Config) ValidateUnknownAttrs(fields schema.Fields, defaults schema.De
...
+			// We know this name isn't in the global fields, or it wouldn't be
+			// an UnknownAttr, it also appears to not be in the extra fields
+			// that are provider specific.  Check to see if an alternative
+			// spelling is in either the extra fields or the core fields.
 			if val, isString := value.(string); isString && val != "" {
 				// only warn about attributes with non-empty string values
-				logger.Warningf("unknown config field %q", name)
+				altName := strings.Replace(name, "_", "-", -1)
+				if extrafields[altName] != nil || fields[altName] != nil {
+					logger.Warningf("unknown config field %q, did you mean %q", name, altName)
```
Comment (wupeka) [inline environs/config/config.go]: ..did you mean %q?"

**Comment by jameinel** [comment]
$$merge$$ thanks!

**Comment by jameinel** [comment]
[xenial] Setting up golang-1.8-race-detector-runtime (0.0+svn285455~ubuntu16.04.1) ... [xenial] Processing triggers for libc-bin (2.23-0ubuntu9) ... [xenial] + go env [xenial] -bash: line 59: go: command not found [xenial] + EXIT_STATUS=127

go not found doesn't seem like a bug in my patch

$$merge$%

**Comment by jameinel** [comment]
$$merge$$

**Comment by jameinel** [comment]
$$merge$$

**Comment by jameinel** [comment]
$$merge$$

**Comment by wupeka** [comment]
$$merge$$

**Pull request merged**